common/src/main/java/org/schemarepo/BaseRepository.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.schemarepo;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Base class for all repository implementations.
+ */
+public abstract class BaseRepository implements Repository {
+
+  protected Logger logger = LoggerFactory.getLogger(getClass());
+
+  protected boolean closed;
+
+  /**
+   * Asserts the repository is in a valid state (for ex. not closed).
+   * @throws java.lang.IllegalStateException if the repository is NOT in a valid state
+   */
+  public void isValid() {
+    if (closed) {
+      throw new IllegalStateException("Repository is closed");
+    }
+  }
+
+  @Override
+  public void close() throws IOException {
+    logger.info("Closing {}", this);
+  }
+
+  /**
+   * Expose certain configuration elements as properties.
+   * This can be used to construct a useful toString() response, for ex.
+   * <p>Remember to call <pre>super</pre> when overriding!</p>
+   * @param properties Properties object to populate
+   */
+  protected void exposeConfiguration(final Map<String, String> properties) {
```

Comment:
I see your point. I prefer to use immutable objects and final variables as often as possible, and java more often than not tends to yield fairly bulky code esp. given that we're trying to avoid external dependencies such as Guava collections. I got lax here. Refactoring.